### PR TITLE
[GEN] Replace GenISA usages with OCL for `2Dblockstore`

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

File: test/TritonGEN/tritongen-to-llvm.mlir [hunk 1]
```diff
@@ -287,26 +287,3 @@ llvm.func @triton_gen.dpas.f32(%c : vector<8xf32>, %a : vector<4xf32>, %b : vect
   %0 = triton_gen.dpas %c, %a, %b {pa = tf32, pb = tf32, rc = 8} : (vector<8xf32>, vector<4xf32>, vector<8xf32>) -> vector<8xf32>
   llvm.return
 }
-
-// -----
-
-// CHECK:  llvm.func spir_funccc @llvm.genx.GenISA.LSC2DBlockWrite.v8f32(i64, i32, i32, i32, i32, i32, i32, i32, i32, i32, i1, i1, i32, vector<8xf32>)
-
-llvm.func @triton_gen.2Dblockstore(%ptr : !llvm.ptr, %base_width : i32, %base_height : i32, %base_pitch : i32, %x : i32, %y : i32, %stored_val : vector<8xf32>) {
```
Comment (whitneywhtsang) [inline test/TritonGEN/tritongen-to-llvm.mlir:295]: Moved to `test/TritonGEN/tritongen-2Dblockstore-to-llvm.mlir`.

**Review by victor-eds — COMMENTED** [review]
LGTM. Can we have a test case with non-`Default` `cache_control`?

**Comment by whitneywhtsang** [comment]
> LGTM. Can we have a test case with non-`Default` `cache_control`?

Added.

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 2]
```diff
@@ -507,88 +507,114 @@ createBlock2DReadWithAddressPayloadUpdate(TritonGEN::Matrix2DBlockLoadOp op,
...
+  SmallVector<Value> args{op.getPtr(),        op.getBaseWidth(),
+                          op.getBaseHeight(), op.getBasePitch(),
+                          byteCoord,          storeValPtr};
+
+  intel::AttrBuilder funcAttrBuilder(*context);
+  intel::AttrBuilder param0AttrBuilder(*context);
+  intel::AttrBuilder param5AttrBuilder(*context);
+  funcAttrBuilder.addPassthroughAttribute(llvm::Attribute::NoUnwind);
+  param0AttrBuilder.addAttribute(llvm::Attribute::NonNull);
+  param0AttrBuilder.addAttribute(llvm::Attribute::WriteOnly);
```
[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:616]: Confirmed no performance degradation.

**Review by victor-eds — APPROVED** [review]

**Comment by etiotto** [comment]
This is causing backward compatibility problems with code that relies on the GenISA API. @whitneywhtsang let's revert the PR please.

**Comment by whitneywhtsang** [comment]
> This is causing backward compatibility problems with code that relies on the GenISA API. @whitneywhtsang let's revert the PR please.

Created https://github.com/intel/intel-xpu-backend-for-triton/pull/1487 so we can use GenISA control by env var in the meantime.

**Pull request merged**